GVRf/Framework/framework/src/main/java/org/gearvrf/GVRContext.java

```diff
@@ -193,6 +194,19 @@ public GVRActivity getActivity() {
     }
 
     /**
+     * Get the {@link GVRImporter} to use for loading assets.
+     * 
+     * A {@code GVRImporter} loads models asynchronously from your applications's
+     * local storage or the network.
+     * 
+     * 
+     * @return The asset loader associated with this context.
+     */
+    public GVRImporter getAssetLoader() {
```

Comment:
Yes, we do. The preferred way will be to use the asset loader. For ALL assets, even textures. I took a deep look into the texture loading issues because we have some bugs there. I have a plan to fix it by making the whole texture loading thing a lot simpler but that will take a while.

So I am doing it in several stages. The first stage is handling the model loading. If we use the new API which waits for all the textures to be loaded  before adding the model to the scene, it will fix some of the problems.

The second stage will be to move texture loading from GVRContext to the asset loader. This is a lot harder because I don’t want to just propagate the existing API. I would like to make something cleaner that will immediately return a GVRTexture (not a future texture or any callback thing). This texture will “magically” load behind the scenes without the user having to do anything. When all the textures for a scene object actually have data present, the lower level layer will render it. Of course, this requires changes to the C++ layer (lots of them) and changes to the implementation of the Java API. I have started this work and will release that in stages too. It is in the “texturereform” branch of my personal repo if you want to look at it.

The final stage will be to change all the samples to use the newer Java API and completely deprecate the old one.

Be patient and it will get less confusing…

From: rahul [mailto:notifications@github.com]
Sent: Thursday, June 2, 2016 5:51 PM
To: Samsung/GearVRf <GearVRf@noreply.github.com>
Cc: Nola Donato <nola.donato@samsung.com>; Author <author@noreply.github.com>
Subject: Re: [Samsung/GearVRf] Wait for textures to be loaded before adding model to the scene (#597)


In GVRf/Framework/framework/src/main/java/org/gearvrf/GVRContext.java<https://github.com/Samsung/GearVRf/pull/597#discussion_r65643197>:

> @@ -193,6 +194,19 @@ public GVRActivity getActivity() {

>      }

>

>      /**

> +     * Get the {@link GVRImporter} to use for loading assets.

> +     *

> +     * A {@code GVRImporter} loads models asynchronously from your applications's

> +     * local storage or the network.

> +     *

> +     *

> +     * @return The asset loader associated with this context.

> +     */

> +    public GVRImporter getAssetLoader() {

Also, there seems to be two conflicting ways of getting an importer object - using this call getAssetLoader or creating one using GVRImporter importer = new GVRImporter(context). This might be a little too confusing. We need to decide on one or the other.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Samsung/GearVRf/pull/597/files/fb58b0721681e130ce48eec2b79e1fc44b7ecf78#r65643197>, or mute the thread<https://github.com/notifications/unsubscribe/AQw1S55ZvuCrFezBg4ckJ8UoBo67ydRzks5qH3qBgaJpZM4ItA3y>.